### PR TITLE
fix: add back missing MeshTrace doc

### PR DESCRIPTION
**Pull request by lahabana**
It got lost as part of the replatform.
Also improved the docs with more examples and more info.

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### CODE REVIEW

File: app/docs/dev/policies/meshtrace.md [hunk 1]
```diff
@@ -1,23 +1,23 @@
 ---
-title: Mesh Trace (beta)
+title: MeshTrace (beta)
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:2]:
```suggestion
title: MeshTrace 
beta: true
```
I think now that we have Jekyll, you might be able to use this frontmatter to get a "beta" badge to display on the page: https://docs.konghq.com/contributing/markdown-rules/#markdown-front-matter

File: app/docs/dev/policies/meshtrace.md [hunk 2]
```diff
@@ -42,17 +42,96 @@ For HTTP you can also manually forward the following headers:
...
-:::::::: tabs :options="{ useUrlFragment: false }"
-::::::: tab "Zipkin"
-:::::: tabs :options="{ useUrlFragment: false }"
-::::: tab "Kubernetes"
-:::: tabs :options="{ useUrlFragment: false }"
-::: tab "Minimal example"
+### Sampling
+
+{% tip %}
+Most of the time setting only `overall` is sufficient; `random` and `client` are for advanced use cases.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:52]:
```suggestion
Most of the time setting only `overall` is sufficient. `random` and `client` are for advanced use cases.
```
Could we specify what those use cases are? Or are those covered in the links below?

File: app/docs/dev/policies/meshtrace.md [hunk 3]
```diff
@@ -42,17 +42,96 @@ For HTTP you can also manually forward the following headers:
...
+    value: 80
+  random:
+    value: 60
+  client:
+    value: 40
+```
+
+### Tags
+
+You can add tags to trace metadata by directly supplying the value ("literal") or by taking it from a header ("header").
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:77]:
```suggestion
You can add tags to trace metadata by directly supplying the value (`literal`) or by taking it from a header (`header`).
```

File: app/docs/dev/policies/meshtrace.md [hunk 4]
```diff
@@ -42,17 +42,96 @@ For HTTP you can also manually forward the following headers:
...
+  - name: env
+    header:
+      name: x-env
+      default: prod
+  - name: version
+    header:
+      name: x-version
+```
 
+For `header` - if a value is missing then `default` will be used.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:94]:
```suggestion
If a value is missing for `header`, `default` is used.
```

File: app/docs/dev/policies/meshtrace.md [hunk 5]
```diff
@@ -42,17 +42,96 @@ For HTTP you can also manually forward the following headers:
...
+If `default` is not provided then the tag won't be added.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:95]:
```suggestion
If `default` isn't provided, then the tag won't be added.
```

File: app/docs/dev/policies/meshtrace.md [hunk 6]
```diff
@@ -294,72 +357,92 @@ spec:
...
+{% endtabs %}
 
-### Sampling
+### Targeting parts of the infrastructure
 
-:::tip
-Most of the time setting only `overall` is sufficient; `random` and `client` are for advanced use cases.
-:::
+While most commonly we want all the traces to be sent to the same tracing backend,
+we can target parts of a `Mesh` by using finer-grained `targetRef` and a designated backend to trace different paths of our service traffic.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:368]:
```suggestion
you can target parts of a `Mesh` by using a finer-grained `targetRef` and a designated backend to trace different paths of our service traffic.
```

File: app/docs/dev/policies/meshtrace.md [hunk 7]
```diff
@@ -294,72 +357,92 @@ spec:
...
+{% endtab %}
+{% endtabs %}
 
-### Sampling
+### Targeting parts of the infrastructure
 
-:::tip
-Most of the time setting only `overall` is sufficient; `random` and `client` are for advanced use cases.
-:::
+While most commonly we want all the traces to be sent to the same tracing backend,
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:367]:
```suggestion
While usually you want all the traces to be sent to the same tracing backend,
```

File: app/docs/dev/policies/meshtrace.md [hunk 8]
```diff
@@ -294,72 +357,92 @@ spec:
...
+we can target parts of a `Mesh` by using finer-grained `targetRef` and a designated backend to trace different paths of our service traffic.
+This is especially useful when we want traces to never leave a world region, or a cloud, for example.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:369]:
```suggestion
This is especially useful when you want traces to never leave a world region, or a cloud, for example.
```

File: app/docs/dev/policies/meshtrace.md [hunk 9]
```diff
@@ -294,72 +357,92 @@ spec:
...
 
-You can configure sampling settings equivalent to Envoy's:
+In this example we have two zones `east` and `west`, each of these with their own zipkin collector: `east.zipkincollector:9411/api/v2/spans` and `west.zipkincollector:9411/api/v2/spans`.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:371]:
```suggestion
In this example, we have two zones `east` and `west`, each of these with their own Zipkin collector: `east.zipkincollector:9411/api/v2/spans` and `west.zipkincollector:9411/api/v2/spans`.
```

File: app/docs/dev/policies/meshtrace.md [hunk 10]
```diff
@@ -294,72 +357,92 @@ spec:
...
+We want dataplane proxies in each zone to only send traces to their local collector.
 
-- [overall](https://www.envoyproxy.io/docs/envoy/v1.22.5/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto.html?highlight=overall_sampling#extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing)
-- [random](https://www.envoyproxy.io/docs/envoy/v1.22.5/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto.html?highlight=random_sampling#extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing)
-- [client](https://www.envoyproxy.io/docs/envoy/v1.22.5/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto.html?highlight=client_sampling#extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing)
+To do this we use a `TargetRef` kind value of `MeshSubset` to filter which dataplane proxy a policy applies to.
```
Comment (cloudjumpercat) [inline app/docs/dev/policies/meshtrace.md:374]:
```suggestion
To do this, we use a `TargetRef` kind value of `MeshSubset` to filter which dataplane proxy a policy applies to.
```

File: app/docs/dev/policies/meshtrace.md (hunk 1 above)
Comment (mheap) [reply app/docs/dev/policies/meshtrace.md:2]: This won't work unless the template is configured to render a beta badge (which it's not currently)

Comment (lahabana) [reply app/docs/dev/policies/meshtrace.md:2]: https://github.com/kumahq/kuma-website/issues

File: app/docs/dev/policies/meshtrace.md (hunk 2 above)
Comment (lahabana) [reply app/docs/dev/policies/meshtrace.md:52]: Yes they are covered in the Envoy links.

File: app/docs/dev/policies/meshtrace.md [hunk 11]
```diff
@@ -42,17 +42,96 @@ For HTTP you can also manually forward the following headers:
 | MeshGatewayRoute  | ✅        | ❌  | ❌   |
 | MeshHTTPRoute     | ❌        | ❌  | ❌   |
 
-If you don't understand this table you should read [matching docs](matching#policy-matching)
+If you don't understand this table you should read [matching docs](../matching#policy-matching).
```
Comment (slonka) [inline app/docs/dev/policies/meshtrace.md:45]: Isn't it in the same directory?

Comment (lahabana) [reply app/docs/dev/policies/meshtrace.md:45]: That's how these docs are like now. meshtrace is a path in the URL

**Review by slonka — APPROVED** [review]
Looks ok!

**Pull request merged**